_blog/2018/egress-https.md

```diff
@@ -0,0 +1,173 @@
+
+I am using a Kubernetes cluster with
+[automatic sidecar injection]({{home}}/docs/setup/kubernetes/sidecar-injection.html#automatic-sidecar-injection)
+   enabled, so I will simply use `kubectl` command to deploy _details v2_.
+```bash
+kubectl apply -f samples/bookinfo/kube/bookinfo-details-v2.yaml
+```
+
+The updated architecture of the application now looks as follows:
+<figure><img src="img/bookinfo-details-v2.svg" alt="The Bookinfo Application with details v2" title="The Bookinfo Application with details v2" />
+<figcaption>The Bookinfo Application with details v2</figcaption></figure>
+
+Note that the Google Books web service is outside the Istio Service Mesh whose boundaries are marked by a dashed line.
+
+Now let's direct all the traffic to the version _v2_ of the _details_ microservice, using the following Route Rule:
+
+```bash
+cat <<EOF | istioctl create -f -
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: details-v2
+  namespace: default
+spec:
+  destination:
+    name: details
+  route:
+  - labels:
+      version: v2
+EOF
+```
+
+Let's access the web page of the application, after [determining the ingress IP and Port]({{home}}/docs/guides/bookinfo.html#determining-the-ingress-ip-and-port).
+
+Oops... Instead of the book details we have _Error fetching product details_ message displayed:
+<figure><img src="img/errorFetchingBookDetails.png" alt="Error Fetching Product Details" title="Error Fetching Product Details" />
+<figcaption>Error Fetching Product Details Displayed</figcaption></figure>
+
+
+Good news is that our application did not crash. With a good microservice design we do not have **failure propagation**. In our case the failing _details_ microservice does not cause the _productpage_ microservice to fail. Most of the functionality of the application is still provided despite the failure in the _details_ microservice. We have **graceful service degradation**: as you can see, the reviews and the ratings are displayed correctly, and the application is still useful.
+
+Hmm, so what might have gone wrong? Ah... The answer is that I forgot to enable traffic from inside the mesh to an external service, in this case to the Google Books web service. By default, the Istio sidecar proxies ([Envoy proxies](https://www.envoyproxy.io)) block all the traffic to the outside the cluster. To enable such traffic, we must define an [Egress Rule]({{home}}/docs/reference/config/traffic-rules/egress-rules.html).
+
+### Egress Rule for Google Books Web Service
+No worries, let's define an **Egress Rule** and fix our application.
+```bash
+cat <<EOF | istioctl create -f -
+apiVersion: config.istio.io/v1alpha2
+kind: EgressRule
+metadata:
+  name: googleapis
+  namespace: default
+spec:
+  destination:
+      service: "*.googleapis.com"
+  ports:
+      - port: 443
+        protocol: https
+EOF
+```
```

Comment:
is it possible to have the yaml in git and create some e2e for this?  that can allow us to ensure the step in the blog is always current.

Reply:
Good idea. I will submit a PR to istio/istio, and then fix the blog. It can be done after the blog is published.

Reply:
Added yamls and a smoke e2e test for testing the scenario of the blog:
https://github.com/istio/istio/pull/2923.